core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -583,7 +583,10 @@ class SparkContext(config: SparkConf) extends Logging {
     _applicationId = _taskScheduler.applicationId()
     _applicationAttemptId = _taskScheduler.applicationAttemptId()
     _conf.set("spark.app.id", _applicationId)
-    _applicationAttemptId.foreach(attemptId => _conf.set(APP_ATTEMPT_ID, attemptId))
+    _applicationAttemptId.foreach { attemptId =>
+      _conf.set(APP_ATTEMPT_ID, attemptId)
+      _env.blockManager.blockStoreClient.setAppAttemptId(attemptId.toInt)
```

Comment:
Thinking more, attempt id is an `int` for yarn - we dont have an expectation for other cluster managers - and we currently treat it as a String.
+CC @Ngone51, thoughts of coercing it always to `int` in `BlockStoreClient`  and doing the `toInt` here ?

Reply:
Yeah..I think we missed it previously. The best way is to use String type too across protocols. I'm not sure we'd change it now.  Currently, `toInt` looks fine.

Reply:
Resolving conversation based on comment [here](https://github.com/apache/spark/pull/34018#issuecomment-921930779).